File: backend/app/schemas/user_progress.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime

class UserProgressBase(BaseModel):
    user_id: int
    lesson_id: int

class UserProgressCreate(UserProgressBase):
    pass

class UserProgress(UserProgressBase):
    completed_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/lib/models/user_progress.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_progress.g.dart';

@JsonSerializable()
class UserProgress {
  UserProgress({
    required this.userId,
    required this.lessonId,
    required this.completedAt,
  });

  factory UserProgress.fromJson(Map<String, dynamic> json) => _$UserProgressFromJson(json);
  Map<String, dynamic> toJson() => _$UserProgressToJson(this);

  final int userId;
  final int lessonId;
  final DateTime completedAt;
}
